source/jormungandr/jormungandr/scenarios/helper_classes/transfer.py

```diff
@@ -0,0 +1,143 @@
+# Copyright (c) 2001-2022, Hove and/or its affiliates. All rights reserved.
+#
+# This file is part of Navitia,
+#     the software to build cool stuff with public transport.
+#
+# Hope you'll enjoy and contribute to this project,
+#     powered by Hove (www.hove.com).
+# Help us simplify mobility and open public transport:
+#     a non ending quest to the responsive locomotion way of traveling!
+#
+# LICENCE: This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <http://www.gnu.org/licenses/>.
+#
+# Stay tuned using
+# twitter @navitia
+# channel `#navitia` on riot https://riot.im/app/#/room/#navitia:matrix.org
+# https://groups.google.com/d/forum/navitia
+# www.navitia.io
+from navitiacommon import response_pb2
+import itertools
+import logging
+from jormungandr.street_network.street_network import StreetNetworkPathType
+from jormungandr.utils import PeriodExtremity
+from jormungandr.fallback_modes import FallbackModes
+
+NO_ACCESS_POINTS_PHYSICAL_MODES = (
+    'physical_mode:Bus',
+    'physical_mode:Tramway',
+    'physical_mode:BusRapidTransit',
+    'physical_mode:Coach',
+    'physical_mode:Shuttle',
+)
+
+ACCESS_POINTS_PHYSICAL_MODES = ("physical_mode:RapidTransit",)
+
+NO_ACCESS_POINTS_TRANSFER = set(
+    itertools.product(NO_ACCESS_POINTS_PHYSICAL_MODES, NO_ACCESS_POINTS_PHYSICAL_MODES)
+)
+
+ACCESS_POINTS_TRANSFER = set(
```

Comment:
```suggestion
# if `(physical_mode:A, physical_mode:B) in NO_ACCESS_POINTS_TRANSFER` then it means that a transfer 
# where we get out of a vehicle of  `physical_mode:A` and then get in a vehicle of `physical_mode:B` 
# **will not** go through an access point
NO_ACCESS_POINTS_TRANSFER = set(
    itertools.product(NO_ACCESS_POINTS_PHYSICAL_MODES, NO_ACCESS_POINTS_PHYSICAL_MODES)
)
```suggestion
# if `(physical_mode:A, physical_mode:B) in ACCESS_POINTS_TRANSFER` then it means that a transfer 
# where we get out of a vehicle of  `physical_mode:A` and then get in a vehicle of `physical_mode:B` 
# **may use** an access point
ACCESS_POINTS_TRANSFER = set(
```
I had some trouble understanding what these sets are. Are my comments right ?

Reply:
You are the future OpenAI!!!!